src/libsyntax/parse/attr.rs

```diff
@@ -59,6 +63,20 @@ impl<'a> Parser<'a> {
                     self.bump();
                     just_parsed_doc_comment = true;
                 }
+                token::Lifetime(l)
```

Comment:
I feel sort of surprised to see this error message in *this particular* function. It feels like this could be something that we (optionally) run after invocations to this function, perhaps? Or maybe a wrapper? Why is this function even called in these cases, I guess because of attributes on expressions?

Put another way, it just seems surprising to skip a lifetime *here* since it's not like it's part of a malformed outer attribute.

Reply:
Fair enough. I will rework the code to add a new method that proactively looks for lifetimes everywhere that applies.